dev/archery/archery/integration/datagen.py

```diff
@@ -1113,6 +1163,10 @@ def _temp_path():
 
         generate_nested_case(),
 
+        generate_custom_metadata_case().skip_category('Go')
+                                       .skip_category('Java')
+                                       .skip_category('JS'),
+
```

Comment:
@wesm I have skipped everybody but C++ due to incompatible serialization of custom_metadata. I assumed that this would exclude Java from all test involving custom_metadata, but it seems Java is still trying to test Flight https://github.com/apache/arrow/pull/6556/checks?check_run_id=504218272#step:5:3445

I can add `.skip_category(SKIP_FLIGHT)` but that seems incorrect (C++ <-> C++ should still be testable through flight). Am I missing something?

Reply:
https://github.com/apache/arrow/pull/6556/commits/53dbe796de8317279377578d4bf78d4bd1112591